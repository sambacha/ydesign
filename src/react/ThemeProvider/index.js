import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
