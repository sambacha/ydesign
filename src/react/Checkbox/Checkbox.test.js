import Checkbox from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Checkbox component sanity', () => {
  it('has name', () => {
    expect(Checkbox.displayName).toBe('Checkbox');
  });

  it('matches default snapshot', () => {
    const component = render(<Checkbox />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Checkbox />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
