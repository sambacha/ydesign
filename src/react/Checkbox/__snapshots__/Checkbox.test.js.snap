// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component sanity matches default snapshot 1`] = `
.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4E3FCE;
  opacity: 1;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c3 {
  position: relative;
  display: inline-block;
}

.c3 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c3 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c3 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c3 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c1 {
  cursor: pointer;
}

.c1:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<label
  class="c0 c1"
  color="primary"
  display="flex"
  opacity="1"
>
  <div
    class="c2 c3"
  >
    <input
      color="primary"
      type="checkbox"
    />
    <svg
      class="c4"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="checked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <svg
      class="c5"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="unchecked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    class="c6"
    color="text"
    font-size="1"
    font-weight="3"
  />
</label>
`;

exports[`Checkbox component sanity matches themed snapshot 1`] = `
.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4E3FCE;
  opacity: 1;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c3 {
  position: relative;
  display: inline-block;
}

.c3 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c3 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c3 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c3 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c1 {
  cursor: pointer;
}

.c1:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<label
  class="c0 c1"
  color="primary"
  display="flex"
  opacity="1"
>
  <div
    class="c2 c3"
  >
    <input
      color="primary"
      type="checkbox"
    />
    <svg
      class="c4"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="checked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <svg
      class="c5"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="unchecked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    class="c6"
    color="text"
    font-size="1"
    font-weight="3"
  />
</label>
`;
